pkg/types/service_broker.go

```diff
@@ -32,17 +33,28 @@ const maxNameLength = 255
 // ServiceBroker broker struct
 type ServiceBroker struct {
 	Base
-	Secured     `json:"-"`
-	Strip       `json:"-"`
-	Name        string       `json:"name"`
-	Description string       `json:"description"`
-	BrokerURL   string       `json:"broker_url"`
-	Credentials *Credentials `json:"credentials,omitempty"`
+	Secured       `json:"-"`
+	Strip         `json:"-"`
+	Name          string             `json:"name"`
+	Description   string             `json:"description"`
+	BrokerURL     string             `json:"broker_url"`
+	Credentials   *Credentials       `json:"credentials,omitempty"`
+	Catalog       json.RawMessage    `json:"-"`
+	Services      []*ServiceOffering `json:"-"`
+	LastOperation *Operation         `json:"last_operation,omitempty"`
+}
 
-	Catalog  json.RawMessage    `json:"-"`
-	Services []*ServiceOffering `json:"-"`
+func (e *ServiceBroker) GetTlsConfig() (*tls.Config, error) {
+	var tlsConfig tls.Config
+	if e.Credentials.TLS != nil && e.Credentials.TLS.Certificate != "" && e.Credentials.TLS.Key != "" {
+		cert, err := tls.X509KeyPair([]byte(e.Credentials.TLS.Certificate), []byte(e.Credentials.TLS.Key))
+		if err != nil {
+			return &tls.Config{}, err
+		}
+		tlsConfig.Certificates = []tls.Certificate{cert}
+	}
 
-	LastOperation *Operation `json:"last_operation,omitempty"`
+	return &tlsConfig, nil
```

Comment:
I think we would still rather keep it as is - as in this case it should be safer to return an empty TLS config to avoid problems going onwards, for instance when there're attempts to convert it to string